pkg/events/parse/params_test.go

```diff
@@ -282,7 +282,7 @@ func TestArgVal(t *testing.T) {
 		for _, tt := range tests {
 			t.Run(tt.name, func(t *testing.T) {
 				e := trace.Event{Args: []trace.Argument{tt.arg}}
-				val, err := ArgVal[[]string](&e, tt.name)
+				val, err := ArgVal[[]string](e.Args, tt.name)
```

Comment:
A quick observation about context.go. It is currently being used by event processor and not concurrently. Nevertheless it has two issues regarding concurrency support I believe:

1. Store: Read and Increment should be done together to avoid concurrent increments without a read (minor issue: index would increase more than 1 number, no big deal)
2. Get: The mutex is not held during the entire logic. The event Store() can happen in between the ok assignment and the code return (race).

Unit tests should included a threaded test for Store/Get concurrency:

```go
wg.Add(1)
	go func() {
		for i := 0; i < 10; i++ {
			wg.Add(1)
			go func(i int) {
                Store()
				wg.Done()
			}(i)
		}

		for i := 0; i < 10; i++ {
			wg.Add(1)
			go func(i int) {
				Get()
				wg.Done()
			}(i)
		}

		wg.Done()
	}()

	wg.Wait()
```

This is also true for the `counter.go` tests. It is supposed to be atomic but there isn't a single concurrency test at it. Should be checked for races as well.

Reply:
https://github.com/aquasecurity/tracee/issues/3263

and

https://github.com/aquasecurity/tracee/issues/3264

Opened.